python/pyspark/sql/connect/ml/functions.py

```diff
@@ -0,0 +1,38 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from pyspark.sql.connect.utils import check_dependencies
+
+check_dependencies(__name__)
+
+from pyspark.ml import functions as PyMLFunctions
+
+from pyspark.sql.connect.column import Column
+from pyspark.sql.connect.functions import _invoke_function, _to_col, lit
+
+
+def vector_to_array(col: Column, dtype: str = "float64") -> Column:
+    return _invoke_function("vector_to_array", _to_col(col), lit(dtype))
+
+
+vector_to_array.__doc__ = PyMLFunctions.vector_to_array.__doc__
+
+
+def array_to_vector(col: Column) -> Column:
+    return _invoke_function("array_to_vector", _to_col(col))
+
+
+array_to_vector.__doc__ = PyMLFunctions.array_to_vector.__doc__
```

Comment:
Actually, why don't you remove `python/pyspark/sql/tests/connect/ml/test_connect_ml_function.py`, and run the same doctests here instead? That would be less code but the same API coverage.

You could just add, e.g.)

```python
def _test() -> None:
    import doctest
    from pyspark.sql import SparkSession
    import pyspark.sql.connect.ml.functions
    import sys

    globs = pyspark.sql.connect.ml.functions.__dict__.copy()
    globs["spark"] = (
        PySparkSession.builder.appName("sql.connect.functions tests")
        .remote("local[4]")
        .getOrCreate()
    )
    globs["spark"] = spark

    (failure_count, test_count) = doctest.testmod(
        pyspark.sql.connect.ml.functions,
        globs=globs,
        optionflags=doctest.ELLIPSIS | doctest.NORMALIZE_WHITESPACE,
    )
    spark.stop()
    if failure_count:
        sys.exit(-1)


if __name__ == "__main__":
    _test()
```

Reply:
and add this file name into here https://github.com/apache/spark/blob/master/dev/sparktestsupport/modules.py#L530 so doctests actually run in CI

Reply:
will have a try

but I guess I will still need to add a few dedicated tests at first due to https://github.com/apache/spark/pull/40432#discussion_r1136931940